File: 2022/python/day1/d1_2.py
import logging

logging.basicConfig(level=logging.INFO)

elf_loads = []
with open("input.txt", "r") as f:
    most_cals = 0
    elf_load = 0
    for line in f:
        l = line.strip()
        if len(l) > 0:
            logging.debug(l)
            elf_load += int(l)
        else:
            logging.debug(f"elf load: {elf_load}")
            elf_loads.append(elf_load)
            elf_load = 0

answer = sum(sorted(elf_loads)[-3:])
logging.info(f"answer = {answer}")

File: 2022/lua/day1/d1_2.lua

elf_load = 0
elf_loads = {}
for line in io.lines("/home/joe/dev/advent-of-code/2022/python/day1/input.txt") do
    if line ~= '' then
	elf_load = elf_load + tonumber(line)
    else
	table.insert(elf_loads, elf_load)
	elf_load = 0
    end
end
table.sort(elf_loads)

answer = elf_loads[#elf_loads - 2] + elf_loads[#elf_loads - 1] + elf_loads[#elf_loads]

print(answer)
